foundations/src/telemetry/log/mod.rs

```diff
@@ -43,6 +43,12 @@ pub fn set_verbosity(level: Level) -> Result<()> {
     Ok(())
 }
 
+/// Gets the current log's verbosity.
+pub fn get_verbosity() -> LogVerbosity {
+    let harness = LogHarness::get();
+    harness.settings.verbosity.clone()
```

Comment:
Clippy doesn't like the clone as the verbosity is `Copy`.

```suggestion
    harness.settings.verbosity
```

Note that if you accept my suggestion, you will need to do a rebase and squash so that our history remains nicely bisectable.